config-model-generator/pom.xml

```diff
@@ -67,6 +67,32 @@
                         <argument>${config-model-file}</argument>
                     </arguments>
                 </configuration>
+                <executions>
+                    <execution>
+                        <!-- This execution is included here to generate the required files as part of the regular
+                             Maven build so that a successful `mvn clean install` isn't contingent on running make first.
+                             This helps with two scenarios:
+                                1) Tests failing in cluster-operator because no Kafka configuration existed
+                                2) Tests failing in cluster-operator because config files for now unsupported Kafka versions still exist -->
+                        <id>generate-config-models</id>
+
+                        <!-- Need a phase after compile (so that the class exists), but before test -->
+                        <phase>process-classes</phase>
+                        <goals>
+                            <goal>exec</goal>
+                        </goals>
+
+                        <!-- Use the shell script instead of executing directly, as we need to execute KafkaConfigModelGenerator
+                             once for every supported version, and the existing shell scripts already handle determing what those versions are -->
+                        <configuration>
+                            <executable>${project.basedir}${file.separator}build-config-models.sh</executable>
+                            <arguments>
+                                <argument>build</argument>
+                            </arguments>
+                            <workingDirectory>${project.basedir}</workingDirectory>
+                        </configuration>
+                    </execution>
+                </executions>
```

Comment:
Should `clean` be also done via Maven then? It seems a bit weird to have only one part driven by Maven.

Reply:
Certainly can. However, it looks like this change broke CodeQL, and given it was only a minor annoyance (that I only really noticed because I had to rebuild my laptop), not sure it's worth merging if it's breaking tooling. 

What do you think?

Reply:
You mean to just close this PR and leave it be as it is?

Reply:
Yep, forget about the PR - it's breaking CodeQL because `autobuild` runs with a clean local repo, and only runs to `package`, not `install`, so adding the execution  into the Maven build for `config-model-generator` will actually break this for anyone running anything less than `install` in the Maven lifecycle on a clean repo, which really defeats the purpose of it.  

It's only a minor problem occasionally encountered. A better solution might be to document in the dev guide that running `make java_build` after first cloning the repository ensures a straightforward Maven experience.